controllers/reconcilers/prometheus_installation/prometheus_installation_reconciler.go

```diff
@@ -56,6 +54,25 @@ func (r *Reconciler) Cleanup(ctx context.Context, cr *v1.Observability) (v1.Obse
 		return v1.ResultFailed, err
 	}
 
+	// We have to remove the prometheus operator deployment manually
+	deployments := &appsv1.DeploymentList{}
+	opts := &client.ListOptions{
+		Namespace: cr.Namespace,
+	}
+	err = r.client.List(ctx, deployments, opts)
+	if err != nil {
+		return v1.ResultFailed, err
+	}
+
+	for _, deployment := range deployments.Items {
```

Comment:
@MStokluska Drafting this PR we went back to v3.0.15 on this reconciler which included this. Think it might be redundant as it had been removed post-v3.0.15. Should be fine to remove. Thanks